docs/spec.md

```diff
@@ -4145,17 +4145,17 @@ Afterwards, within each `process_group`:
 
 #### Constraints
 
-* (C1) `dim(operand, scatter_dimension) % dim(process_groups, 1) = 0`.
-* (C2) `0 <= scatter_dimension < rank(operand)`.
-* (C3) `is_unique(replica_groups)`.
-* (C4) `size(replica_groups)` is defined as:
+* (C1) `0 <= scatter_dimension < rank(operand)`.
+* (C2) `is_unique(replica_groups)`.
+* (C3) `size(replica_groups)` is defined as:
   * `num_replicas` if `cross_replica` is used.
   * `num_replicas` if `cross_replica_and_partition` is used.
   * `num_processes` if `flattened_ids` is used.
-* (C5) `0 <= replica_groups < size(replica_groups)`.
-* (C6) If `use_global_device_ids = true`, then `channel_id > 0`.
-* (C7) `computation` has type `(tensor<E>, tensor<E>) -> (tensor<E>)` where
+* (C4) `0 <= replica_groups < size(replica_groups)`.
```

Comment:
Thoughts on adding a constraint for `channel_type == 0`? This doesn't cause any issues in StableHLO-->HLO for XLA compilation, but this value does not round trip properly. Not sure how we define "HLO semantics" like this in term of our spec.

When going HLO-->MHLO in [openxla/xla > hlo_function_importer.cc](https://github.com/openxla/xla/blob/4ed4ae233a904df1d0b085a654ecb162632b401f/xla/translate/hlo_to_mhlo/hlo_function_importer.cc#L2202) this value is set to `0`. This applies to `CollectivePermute, AllGather, AllReduce, AllToAll, ReduceScatter`.

Reply:
channel_type is intentionally not part of the StableHLO spec for collective ops, because it is ignored by XlaBuilder. Here's an example from XlaBuilder::ReduceScatter ([link](https://github.com/tensorflow/tensorflow/blob/4c40c3991e4a71f4819c22f24f4f580893ea121e/tensorflow/compiler/xla/client/xla_builder.cc#L3389-L3391)).